compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsAnydataUniqueVisitor.java

```diff
@@ -278,7 +278,8 @@ public Boolean visit(BRecordType type) {
                 return false;
             }
         }
-        type.isAnyData = type.sealed || isPureTypeUniqueVisitor.visit(type.restFieldType);
+        type.isAnyData =
+                type.sealed || (type.restFieldType != null && isPureTypeUniqueVisitor.visit(type.restFieldType));
```

Comment:
What's the scenario where the `restFieldType` becomes null? Usually, `noType` is set. But if this is while resolving, is this called again later to reset `isAnyData`?

Reply:
Fixed by 67ab648409dba4dece5cb50f655125d24c950828

Reply:
Shall we also add a test for subsequent resolution of `isAnyData` if we don't have one already?